frameworks/mlr3automl/exec.R

```diff
@@ -0,0 +1,66 @@
+library(mlr3)
+library(mlr3learners)
+library(mlr3extralearners)
+library(mlr3tuning)
+library(mlr3pipelines)
+library(mlr3automl)
+library(mlr3oml)
+
+run <- function(train_file, test_file, target.index, type, output_predictions_file, cores, time.budget, meta_results_file, seed, name) {
+  start_time = Sys.time()
+  set.seed(seed)
+  
+  train <- mlr3oml::read_arff(train_file)
+  colnames(train) <- make.names(colnames(train))
+  target <- colnames(train)[target.index]
+
+  test <- mlr3oml::read_arff(test_file)
+  colnames(test) <- make.names(colnames(test))
+  
+  print(paste("Finished loading data after ", Sys.time() - start_time, " seconds"))
+  remaining_budget = as.integer(start_time - Sys.time() + time.budget)
+  print(paste("remaining budget: ", remaining_budget, " seconds"))
+
+  if (type == "classification") {
+    train <- TaskClassif$new("benchmark_train", backend = train, target = target)
+    test <- TaskClassif$new("benchmark_test", backend = test, target = target)
+    if ("twoclass" %in% train$properties) {
+      measure = msr("classif.auc")
+    } else {
+      measure = msr("classif.logloss")
+    }
+  } else if (type == "regression") {
+    train <- TaskRegr$new("benchmark_train", backend = train, target = target)
+    test <- TaskRegr$new("benchmark_test", backend = test, target = target)
+    measure <- msr("regr.rmse")
+  } else {
+    stop("Task type not supported!")
+  }
+
+  training <- function() {
+    model <<- AutoML(train, runtime = as.integer(remaining_budget * 0.8), measure = measure)
+    model$train()
+  }
+
+  train_duration <- system.time(training())[['elapsed']]
+  print(paste("Finished training model after ", difftime(Sys.time(), start_time, units = "secs"), " seconds"))
+
+  prediction <- function() preds <<- model$predict(test)
```

Comment:
if you don't previously define `model` and `preds` in the context of the `run` function, those will be created in the global env, which is usually not desired (you don't know what is in that context).
I guess here it's fine because the R process only executes this `run` function, but fyi, it should usually be avoided.

Reply:
Thanks for the quick response!
I fixed the scoping issues, thanks for the hint